cli/src/commands/unsign.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap_complete::ArgValueCandidates;
+use indexmap::IndexSet;
+use itertools::Itertools;
+use jj_lib::commit::Commit;
+use jj_lib::commit::CommitIteratorExt;
+use jj_lib::signing::SignBehavior;
+
+use super::sign::check_commits_authored_by;
+use crate::cli_util::CommandHelper;
+use crate::cli_util::RevisionArg;
+use crate::command_error::CommandError;
+use crate::complete;
+use crate::ui::Ui;
+
+/// Drop a cryptographic signature
+#[derive(clap::Args, Clone, Debug)]
+pub struct UnsignArgs {
+    /// What revision(s) to unsign
+    #[arg(
+        long, short,
+        default_value = "@",
+        value_name = "REVSETS",
+        add = ArgValueCandidates::new(complete::mutable_revisions),
+    )]
+    revisions: Vec<RevisionArg>,
+    /// Unsign a commit that is not authored by you.
+    #[arg(long, short)]
+    allow_not_mine: bool,
```

Comment:
@necauqua Do we need `--allow-not-mine` for unsign?

Reply:
Actually, it's the same thing with `jj undo` - I now think that we don't need this flag at all in both subcommands.

If jj prints a warning that you messed with someone's signature, you can just undo that.

Or, if we keep it, it's better to have it in both ones for consistency

Reply:
> Actually, it's the same thing with jj undo - I now think that we don't need this flag at all in both subcommands.
>
> If jj prints a warning that you messed with someone's signature, you can just undo that.

Couldn't agree more on removing the flag entirely. That also leaves us with a simpler interface.